buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/publish/PublishingPlugin.groovy

```diff
@@ -121,10 +127,27 @@ class PublishingPlugin implements Plugin<Project> {
                             publisher.decorate(sub, it)
                         }
                     }
+
+                    // Add changed projects to publishChangedToBuildDir.
+                    if (changedProjects.contains(sub)) {
+                        publishChangedToBuildDir.dependsOn "$sub.path:publishMavenAarPublicationToBuildDirRepository"
+                        def dependencies = sub.configurations.implementation.dependencies
+                        for (def dep : dependencies) {
+                            if (dep instanceof ProjectDependency) {
+                                def task = dep.getDependencyProject()
+                                        .tasks.findByName("publishMavenAarPublicationToBuildDirRepository")
+                                if (task != null) {
+                                    publishChangedToBuildDir.dependsOn task
```

Comment:
Why do we need to make `publishChangedToBuildDir` directly depend on `sub`'s dependencies' `publishMavenAarPublicationToBuildDirRepository`? Shouldn't `sub`'s `publishMavenAarPublicationToBuildDirRepository` already depend on it's dependencies' `publishMavenAarPublicationToBuildDirRepository`?

Reply:
That's not how those tasks are configured. They don't publish their dependencies.